e2etests/node_scan_rhcos_test.go

```diff
@@ -137,13 +140,15 @@ func TestGRPCGetRHCOSNodeVulnerabilities(t *testing.T) {
 						Vulnerabilities: []*v1.Vulnerability{},
 					},
 				},
+				NodeNotes: nil,
 			},
 		},
 		{
 			name:    "Uncertified scan is unsupported for RHCOS and returns no features",
 			request: buildRequestCase([]v1.Note{v1.Note_CERTIFIED_RHEL_SCAN_UNAVAILABLE}),
 			responseContains: &v1.GetNodeVulnerabilitiesResponse{
-				Features: []*v1.Feature{},
+				Features:  []*v1.Feature{},
```

Comment:
Given what we have discussed in comments (https://github.com/stackrox/scanner/pull/1004#discussion_r1043601763), this test does not actually validate that the scan returns **no features** (unless I'm reading the code incorrectly).

Basically, the test checks that an empty set is a subset of returned features. By definition, the empty set is a subset of any possible set.

Maybe we over-edited the thing? Either way, some assertions have to be modified now.

Reply:
You are right, this needed a refresh. 
In [6692db4](https://github.com/stackrox/scanner/pull/1004/commits/6692db429eba7737b9c090e679f5fe5dad836a39) I reworked this a bit, so that if we expect 0 features or 0 vulnerabilities, then we use `assert.Equals` on the slice lengths. 
In the other cases we use:
- `assertIsSubset` on vulns
-  each `expected` feature must be found within the `got`, and their length must match.

I know the cleanest solution would be to always assert equality, but this approach could lead to the tests failing in the future when new vulnerabilities are discovered. 

Let me know if that is enough for this case.

Reply:
Well, the problem is that `grep` which currently serves as an example package without vulnerabilities may get vulnerabilities detected in the future. The comparison with `0` will start failing then.
We can either:
- do nothing now and address the issue by changing the package when tests start breaking,
- find a different package that does so little that it will not have vulnerabilities (e.g. in Debian there are meta-packages that have empty contents but bring many transitive dependencies once installed),
- try see how to use a snapshot of vulnerability data in test so that you can be confident that detected vulnerabilities don't change.

I'm fine with any of these options.